app/services/amp/AmpCacheService.scala

```diff
@@ -42,7 +42,7 @@ class AmpCacheService @Inject()(cacheConnector: DataCacheConnector)
 
     for {
       existing <- cacheConnector.fetch[Amp](credId, Amp.key)
-      result   <- cacheConnector.save[Amp](credId, Amp.key, existing.getOrElse(Amp(credId)).data(ampData))
+      result   <- cacheConnector.save[Amp](credId, Amp.key, existing.map(amp=> amp.data(ampData)).getOrElse(Amp().data(ampData)))
```

Comment:
Does this hold if existing is null? 

Used to be `existing.getOrElse(Amp(credId)).data(ampData))` to cater for existing being null. What happens if we map over null and try and set data? Also the removed line is less code? If there is a valid reason for changing this then fair enough.